libweb3jsonrpc/WebThreeStubServerBase.cpp

```diff
@@ -561,91 +756,325 @@ std::string WebThreeStubServerBase::eth_solidity(std::string const& _code)
+	
+	return toJS(client()->installWatch(filter));
+}
+
+bool WebThreeStubServerBase::eth_uninstallFilter(string const& _filterId)
+{
+	unsigned id;
+	
+	try
+	{
+		id = jsToInt(_filterId);
+	}
+	catch (...)
+	{
+		BOOST_THROW_EXCEPTION(JsonRpcException(Errors::ERROR_RPC_INVALID_PARAMS));
+	}
+	
+	// TODO: throw an error if there is no watch with given id?
+	client()->uninstallWatch(id);
+	return true;
+}
+
+Json::Value WebThreeStubServerBase::eth_getFilterChanges(string const& _filterId)
+{
+	unsigned id;
+	
+	try
+	{
+		id = jsToInt(_filterId);
+	}
+	catch (...)
+	{
+		BOOST_THROW_EXCEPTION(JsonRpcException(Errors::ERROR_RPC_INVALID_PARAMS));
+	}
+	
+	// TODO: throw an error if there is no watch with given id?
+	auto entries = client()->checkWatch(id);
+	if (entries.size())
+		cnote << "FIRING WATCH" << id << entries.size();
+	return toJson(entries);
+}
+
+Json::Value WebThreeStubServerBase::eth_getFilterLogs(string const& _filterId)
+{
+	unsigned id;
+	
+	try
+	{
+		id = jsToInt(_filterId);
+	}
+	catch (...)
+	{
+		BOOST_THROW_EXCEPTION(JsonRpcException(Errors::ERROR_RPC_INVALID_PARAMS));
+	}
+	
+	// TODO: throw an error if there is no watch with given id?
+	return toJson(client()->logs(id));
+}
+
+Json::Value WebThreeStubServerBase::eth_getLogs(Json::Value const& _json)
+{
+	LogFilter filter;
+	
+	try
+	{
+		filter = toLogFilter(_json);
+	}
+	catch (...)
+	{
+		BOOST_THROW_EXCEPTION(JsonRpcException(Errors::ERROR_RPC_INVALID_PARAMS));
+	}
+	
+	return toJson(client()->logs(filter));
+}
+
+Json::Value WebThreeStubServerBase::eth_getWork()
 {
-	return client()->number();
+	Json::Value ret(Json::arrayValue);
+	auto r = client()->getWork();
+	ret.append(toJS(r.first));
+	ret.append(toJS(r.second));
+	return ret;
 }
 
-int WebThreeStubServerBase::eth_peerCount()
+bool WebThreeStubServerBase::eth_submitWork(string const& _nonce, string const& _mixHash)
 {
-	return network()->peerCount();
+	
+	Nonce nonce;
+	h256 mixHash;
+	
+	try
+	{
+		nonce = jsToFixed<Nonce::size>(_nonce);
+		mixHash = jsToFixed<32>(_mixHash);
+	}
+	catch (...)
+	{
+		BOOST_THROW_EXCEPTION(JsonRpcException(Errors::ERROR_RPC_INVALID_PARAMS));
+	}
+	
+	return client()->submitWork(ProofOfWork::Proof{nonce, mixHash});
+}
+
+string WebThreeStubServerBase::eth_register(string const& _address)
+{
+	Address address;
+	
+	try
+	{
+		address = jsToAddress(_address);
+	}
+	catch (...)
+	{
+		BOOST_THROW_EXCEPTION(JsonRpcException(Errors::ERROR_RPC_INVALID_PARAMS));
+	}
+	
+	return toJS(m_accounts->addProxyAccount(address));
+}
+
+bool WebThreeStubServerBase::eth_unregister(string const& _accountId)
+{
+	unsigned id;
+	
+	try
+	{
+		id = jsToInt(_accountId);
+	}
+	catch (...)
+	{
+		BOOST_THROW_EXCEPTION(JsonRpcException(Errors::ERROR_RPC_INVALID_PARAMS));
+	}
+
+	// TODO: throw an error on no account with given id
+	return m_accounts->removeProxyAccount(id);
+}
+
+Json::Value WebThreeStubServerBase::eth_queuedTransactions(string const& _accountId)
```

Comment:
fetchQueuedTransactions maybe a good name (with the consideration that a side effect of this method is clearing of the queue)